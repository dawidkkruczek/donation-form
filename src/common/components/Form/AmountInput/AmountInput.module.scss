.root {
  position: relative;
}

.icon {
  color: var(--color-midnight-gray);
  height: 24px;
  left: 8px;
  position: absolute;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  width: 24px;
}

.input {
  padding-left: 40px;
}
