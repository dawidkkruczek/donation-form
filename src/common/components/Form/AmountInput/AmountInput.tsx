import React, { ReactElement } from 'react';
import { NumericFormat } from 'react-number-format';
import classNames from 'classnames';
import Input, { Props as InputProps } from 'common/components/Form/Input/Input';
import styles from './AmountInput.module.scss';
import { ReactComponent as IconComponent } from './assets/icon.svg';

// override some of InputProps to match NumericFormat props
export interface Props extends InputProps {
  defaultValue?: string;
  type?: 'text';
  value?: string;
}

function AmountInput({ className, ...inputProps }: Props): ReactElement {
  return (
    <div className={classNames(styles.root, className)}>
      <IconComponent className={styles.icon} />

      <NumericFormat
        allowNegative={false}
        className={styles.input}
        customInput={Input}
        data-testid="amount-input-input"
        decimalScale={2}
        fixedDecimalScale
        placeholder="0.00"
        thousandSeparator
        {...inputProps}
      />
    </div>
  );
}

export default AmountInput;
