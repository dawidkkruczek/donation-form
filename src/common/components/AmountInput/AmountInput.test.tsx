import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import AmountInput from './AmountInput';

describe('common/components/AmountInput', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<AmountInput />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should format value correctly', () => {
    const { getByPlaceholderText } = render(<AmountInput />);
    const input = getByPlaceholderText('0.00') as HTMLInputElement;

    fireEvent.change(input, { target: { value: '1000' } });
    expect(input.value).toBe('1,000');

    fireEvent.change(input, { target: { value: '1000.00' } });
    expect(input.value).toBe('1,000.00');
  });
});
