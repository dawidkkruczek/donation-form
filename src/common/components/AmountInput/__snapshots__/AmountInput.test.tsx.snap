// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/components/AmountInput should render correctly 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <svg
      class="icon"
    >
      icon.svg
    </svg>
    <input
      class="root input"
      inputmode="numeric"
      placeholder="0.00"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
